packages/tracing-internal/src/browser/browserTracingIntegration.ts

```diff
@@ -0,0 +1,529 @@
+          if (activeSpan) {
+            DEBUG_BUILD && logger.log(`[Tracing] Finishing current transaction with op: ${spanToJSON(activeSpan).op}`);
+            // If there's an open transaction on the scope, we need to finish it before creating an new one.
+            activeSpan.end();
+          }
+          activeSpan = _createRouteTransaction(context);
+        });
+      }
+
+      if (options.instrumentPageLoad && client.emit && shouldUseDefaultPageLoadSpan) {
+        const context: StartSpanOptions = {
+          name: WINDOW.location.pathname,
+          // pageload should always start at timeOrigin (and needs to be in s, not ms)
+          startTimestamp: browserPerformanceTimeOrigin ? browserPerformanceTimeOrigin / 1000 : undefined,
+          op: 'pageload',
+          origin: 'auto.pageload.browser',
+          metadata: { source: 'url' },
+        };
+        startBrowserTracingPageLoadSpan(context);
+      }
+
+      if (options.instrumentNavigation && client.emit) {
+        addHistoryInstrumentationHandler(({ to, from }) => {
+          /**
+           * This early return is there to account for some cases where a navigation transaction starts right after
+           * long-running pageload. We make sure that if `from` is undefined and a valid `startingURL` exists, we don't
+           * create an uneccessary navigation transaction.
+           *
+           * This was hard to duplicate, but this behavior stopped as soon as this fix was applied. This issue might also
+           * only be caused in certain development environments where the usage of a hot module reloader is causing
+           * errors.
+           */
+          if (from === undefined && startingUrl && startingUrl.indexOf(to) !== -1) {
+            startingUrl = undefined;
+            return;
+          }
+
+          if (from !== to) {
+            startingUrl = undefined;
+            // We check this in here again, as a custom instrumentation may have been triggered in the meanwhile
+            if (shouldUseDefaultNavigationSpan) {
+              const context: StartSpanOptions = {
+                name: WINDOW.location.pathname,
+                op: 'navigation',
+                origin: 'auto.navigation.browser',
+                metadata: { source: 'url' },
+              };
+
+              startBrowserTracingNavigationSpan(context);
+            }
+          }
+        });
+      }
+
+      if (markBackgroundSpan) {
+        registerBackgroundTabDetection();
+      }
+
+      if (_experiments.enableInteractions) {
+        registerInteractionListener(options, latestRouteName, latestRouteSource);
+      }
+
+      instrumentOutgoingRequests({
+        traceFetch,
+        traceXHR,
+        tracePropagationTargets,
+        shouldCreateSpanForRequest,
+        enableHTTPTimings,
+      });
+    },
+    // TODO v8: Remove this again
+    // This is private API that we use to fix converted BrowserTracing integrations in Next.js & SvelteKit
+    options,
```

Comment:
Could not think of a better way to allow us to solve this 😬 I'd say it's OK for now, we can remove it in v8.

Reply:
We need to refactor how options works in general with `BrowserTracing`, there's some duplication around setting defaults that I think we can improve.